autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/importing/InBrowserImportPromo.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2025 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.impl.importing
+
+import com.duckduckgo.autofill.api.AutofillFeature
+import com.duckduckgo.autofill.impl.store.InternalAutofillStore
+import com.duckduckgo.autofill.impl.store.NeverSavedSiteRepository
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.withContext
+
+interface InBrowserImportPromo {
+    suspend fun canShowPromo(
+        credentialsAvailableForCurrentPage: Boolean,
+        url: String?,
+    ): Boolean
+}
+
+@ContributesBinding(AppScope::class)
+class RealInBrowserImportPromo @Inject constructor(
+    private val autofillStore: InternalAutofillStore,
+    private val dispatchers: DispatcherProvider,
+    private val neverSavedSiteRepository: NeverSavedSiteRepository,
+    private val autofillFeature: AutofillFeature,
+) : InBrowserImportPromo {
+
+    override suspend fun canShowPromo(
+        credentialsAvailableForCurrentPage: Boolean,
+        url: String?,
+    ): Boolean {
+        return withContext(dispatchers.io()) {
+            if (credentialsAvailableForCurrentPage) {
+                return@withContext false
+            }
+
```

Comment:
Q: are we missing the capability checks here?

Reply:
indeed, thanks. fixed.